src/System.Runtime.Extensions/tests/System/StringComparer.netcoreapp.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Xunit;
+
+namespace System.Tests
+{
+    public static partial class StringComparerTests
+    {
+        public static readonly object[][] FromComparison_TestData =
+        {
+            //             StringComparison                             StringComparer
+            new object[] { StringComparison.CurrentCulture,             StringComparer.CurrentCulture },
+            new object[] { StringComparison.CurrentCultureIgnoreCase,   StringComparer.CurrentCultureIgnoreCase },
+            new object[] { StringComparison.InvariantCulture,           StringComparer.InvariantCulture },
+            new object[] { StringComparison.InvariantCultureIgnoreCase, StringComparer.InvariantCultureIgnoreCase },
+            new object[] { StringComparison.Ordinal,                    StringComparer.Ordinal },
+            new object[] { StringComparison.OrdinalIgnoreCase,          StringComparer.OrdinalIgnoreCase },
+        };
+
+        [Theory]
+        [MemberData(nameof(FromComparison_TestData))]
+        public static void FromComparisonTest(StringComparison comparison, StringComparer comparer)
+        {
+            Assert.Equal(comparer, StringComparer.FromComparison(comparison));
```

Comment:
I've just tested this, which resulted in a `Xunit.Sdk.SameException` when using it with `CurrentCulture` or `CurrentCultureIgnoreCase`. It seems the property always returns a new instance: https://github.com/dotnet/coreclr/blob/1a32d7f2c1be31741c3e71fe35eb2fd22664d12c/src/mscorlib/shared/System/StringComparer.cs#L38
I'm not certain why it was implemented that way, but I don't think it should be tested here using Assert.Same.